File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
       HashMap<String, List<String>> map = new HashMap<>();
       List<List<String>> ans = new ArrayList<>();
       for(String s: strs){
            char ch[] = s.toCharArray();
            Arrays.sort(ch);
            String temp = new String(ch);
            if(map.containsKey(temp)){
               map.get(temp).add(s);
            }
            else{
                List<String> list = new ArrayList<>();
                list.add(s);
                map.put(temp,list);
            }
       }
       for(List<String> l : map.values()){
            ans.add(l);
       }   
       return ans;
    }
}

File: 49-group-anagrams/group-anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        ans = []
        d = {}
        for str in strs:
            l = tuple(sorted(str))
            if l in d:
                d[l].append(str)
            else:
                d[l] = [str]
        for value in d.values():
            ans.append(value)
        return ans
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        